### PR TITLE
[FEATURE] DataAssistantResult Includes Underlying Profiler Execution Time

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Update the `measure_execution_time` utility (developed a significant time ago) to call back with results (not just print).
* Use `@measure_execution_time` on running profiler for `DataAsset` classes.
* Include assertion on `DataAssistantResult.execution_time` in test of `VolumeDataAssistant` subclass.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-735/GREAT-783
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/util.py [hunk 1]
```diff
@@ -175,27 +175,41 @@ def profile_function_call(*args, **kwargs) -> Any:
...
-    pretty_print: bool = False,
-) -> Callable:
+def measure_execution_time(pretty_print: bool = False) -> Callable:
     def execution_time_decorator(func: Callable) -> Callable:
-        func.execution_duration_milliseconds = 0
-
         @wraps(func)
         def compute_delta_t(*args, **kwargs) -> Any:
-            time_begin: int = int(round(time.time() * 1000))
+            """
```
Comment (Shinnnyshinshin) [inline great_expectations/util.py:186]: Thanks for the super-helpful docstring :heart:

File: great_expectations/rule_based_profiler/data_assistant/data_assistant.py [hunk 2]
```diff
@@ -326,3 +321,33 @@ def get_expectation_configurations(self) -> List[ExpectationConfiguration]:
             List of "ExpectationConfiguration" objects, computed by "RuleBasedProfiler" state.
         """
         return self.profiler.get_expectation_configurations()
+
+
+@measure_execution_time(pretty_print=False)
```
Comment (Shinnnyshinshin) [inline great_expectations/rule_based_profiler/data_assistant/data_assistant.py]: :bow: thank you @alexsherstinsky

File: tests/rule_based_profiler/data_assistant/test_volume_data_assistant.py [hunk 3]
```diff
@@ -294,3 +294,5 @@ def test_get_metrics(
     ) == deep_filter_properties_iterable(
         properties=expected_rule_based_profiler_config.to_json_dict()
     )
+
+    assert 0.0 < result.execution_time <= 1.0
```
Comment (Shinnnyshinshin) [inline tests/rule_based_profiler/data_assistant/test_volume_data_assistant.py]: Could we add an inline comment here saying that `execution_time` is seconds?

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM. Thank you sir!

File: tests/rule_based_profiler/data_assistant/test_volume_data_assistant.py (hunk 3 above)
Comment (alexsherstinsky) [reply tests/rule_based_profiler/data_assistant/test_volume_data_assistant.py]: @Shinnnyshinshin Done!  Thanks.

**Pull request merged**